newIDE/app/src/EventsSheet/SearchPanel.js

```diff
@@ -124,7 +131,10 @@ export default class SearchPanel extends PureComponent<Props, State> {
               disabled={!searchText}
               primary
               label={<Trans>Search</Trans>}
-              onClick={this.launchSearch}
+              onClick={() => {
+                this.launchSearch();
+                onGoToNextSearchResult();
```

Comment:
Should this be called even if we don't have result already? (not sure, to be checked :))

Reply:
Bit confused (As in did not get what you said ). This is to iterate through the matched searches on the click of search button.

Reply:
This is the search button. When you click on it for the first time, you:
- launch the search (this is fine)
- then you go to the next search result <-- is this something that we want to do even if you just click on search for the first time.

Reply:
I think you are right, we don't want it to go to the next search on the first click, it might be wrongly placed.
Here the search button has 2 functionalities: 
1) Search for the text.
2) Move to next found text.
So, when the button is clicked the first time, it would search for the text. But for the subsequent button presses it should move to the next search. But we might have a roadblock. What if the text to be searched is changed? We need to reset the button press count and again do (1).

Reply:
> But we might have a roadblock. What if the text to be searched is changed? We need to reset the button press count and again do (1).

Exactly. This means that there must be maybe in the state a boolean or something like this to indicate if we've ran the search once or not - and properly reset this boolean when needed.